File: exJava/quiz2/Q23.java
import java.util.*;

public class Q23 {
  public static void main(String args[]) {
    new Q23Main();
  }
}

class Q23Main {

  public Q23Main() {
    MultiCollection multiCol = new MultiCollection();

    multiCol.addList("AAA");
    multiCol.addList("BBB");
    multiCol.addList("CCC");

    multiCol.addMap("X", 100);
    multiCol.addMap("Y", 200);
    multiCol.addMap("Z", 300);
    
    multiCol.showAll();
  }
}


class MultiCollection {

  ReverseList rlist = new ReverseList();

  public void addList(Object str) {
    rlist.add(str);
  }

  public void addMap(Object key, Object value) {


  }

  public void showAll() {
    System.out.println("## List ##");
    showList(rlist.head);

    System.out.println("## Map ##");

  }

  public void showList(Node n) {
    if (n == null) return;
    showList(n.next);
    System.out.println(n);
  }
}


class ReverseList {
  Node head;

  void add(Object obj){
    Node node = new Node(obj);
    node.next = head;
    head = node;
  }

  Object get() {
    if (head == null) return null;
    Object obj = head;
    head = head.next;
    return obj;
  }
}

class Node {
  Node next;
  Object obj;

  public Node(Object obj) {
    next = null;
    this.obj = obj;
  }

  public String toString() {
    return String.valueOf(obj);
  }
}

File: shibaura-institute-technology/exJava/go/Q23.go
package main

import "fmt"
import "container/list"

func main() {
  mc := newMultiColl()
  
  mc.addList("AAA")
  mc.addList("BBB")
  mc.addList("CCC")

  mc.addMap("X", 100)
  mc.addMap("Y", 200)
  mc.addMap("Z", 300)

  mc.showAll()
}

type multiCollection struct {
  l list.List
  m map[string]int
}
func newMultiColl () (mc *multiCollection) {
  mc = new(multiCollection)
  mc.m = make(map[string]int)
  return 
}
func (mc *multiCollection) addList(str string) {
  mc.l.PushBack(str);
}
func (mc *multiCollection) addMap(str string, value int) {
  mc.m[str] = value
}
func (mc *multiCollection) showAll() {

  fmt.Printf("\n## List ##\n")
  for e := mc.l.Front(); e != nil; e = e.Next() {
    fmt.Printf("%s\n", e.Value)
  }

  fmt.Printf("\n## Map ##\n")
  for key, value := range mc.m {
    fmt.Printf("%s: %d\n", key, value)
  }
}

